wgpu-native/src/device.rs

```diff
@@ -2112,6 +2113,25 @@ pub extern "C" fn wgpu_device_destroy(device_id: DeviceId) {
     gfx_select!(device_id => device_destroy(device_id))
 }
 
+pub fn device_describe<B: GfxBackend>(device_id: DeviceId) -> RawStringMut {
+    let hub = B::hub();
+    let mut token = Token::root();
+    let (adapter_guard, mut token) = hub.adapters.read(&mut token);
+    let (device_guard, _) = hub.devices.read(&mut token);
+    let device = &device_guard[device_id];
+    let adapter = &adapter_guard[device.adapter_id];
+    let adapter_info = &adapter.raw.info;
+    std::ffi::CString::new(adapter_info.name.clone()).unwrap_or_else(|_|
+        std::ffi::CString::new("Adapter name contains a nul byte").unwrap()
+    ).into_raw()
+}
+
+#[cfg(not(feature = "remote"))]
+#[no_mangle]
+pub extern "C" fn wgpu_device_describe(device_id: DeviceId) -> RawStringMut {
```

Comment:
there is no need to go through the C API here: only the stuff in WebGPU upstream spec (Web or native) needs to be exposed via FFI. Everything else that wgpu-rs needs can just be exposed as regular Rust APIs (since they are linked as regular Rust libraries).

Reply:
Ah, got it. I rewrote this to just use regular Rust.